cirq-rigetti/cirq_rigetti/circuit_sweep_executors.py

```diff
@@ -0,0 +1,288 @@
+
+
+def _execute_and_read_result(
+    quantum_computer: QuantumComputer,
+    executable: QuantumExecutable,
+    measurement_id_map: Dict[str, str],
+    resolver: cirq.ParamResolverOrSimilarType,
+    memory_map: Optional[Dict[str, Union[int, float, Sequence[int], Sequence[float]]]] = None,
+) -> cirq.Result:
+    """Execute the `pyquil.api.QuantumExecutable` and parse the measurements into
+    a `cirq.Result`.
+
+    Args:
+        quantum_computer: The `pyquil.api.QuantumComputer` on which to execute
+            and from which to read results.
+        executable: The fully compiled `pyquil.api.QuantumExecutable` to run.
+        measurement_id_map: A dict mapping cirq measurement keys to pyQuil
+            read out regions.
+        resolver: The `cirq.ParamResolverOrSimilarType` to include on
+            the returned `cirq.Result`.
+        memory_map: A dict of values to write to memory values on the
+            `quantum_computer`. The `pyquil.api.QuantumAbstractMachine` reads these
+            values into memory regions on the pre-compiled `executable` during execution.
+
+    Returns:
+        A `cirq.Result` with measurements read from the `quantum_computer`.
+
+    Raises:
+        ValueError: measurement_id_map references an undefined pyQuil readout region.
+    """
+    if memory_map is None:
+        memory_map = {}
+
+    for region_name, values in memory_map.items():
+        executable.write_memory(region_name=region_name, value=values)
+    qam_execution_result = quantum_computer.qam.run(executable)
+
+    measurements = {}
+    # For every key, value in QuilOutput#measurement_id_map, use the value to read
+    # Rigetti QCS results and assign to measurements by key.
+    for cirq_memory_key, pyquil_region in measurement_id_map.items():
+        readout = qam_execution_result.readout_data.get(pyquil_region)
+        if readout is None:
+            raise ValueError(f'readout data does not have values for region "{pyquil_region}"')
+        measurements[cirq_memory_key] = readout
+    logger.debug(f"measurement_id_map {measurement_id_map}")
+    logger.debug(f"measurements {measurements}")
+
+    # collect results in a cirq.Result.
+    result = cirq.Result(
+        params=cast(cirq.ParamResolver, resolver or cirq.ParamResolver({})),
+        measurements=measurements,
+    )  # noqa
+    return result
+
+
+def _get_param_dict(resolver: cirq.ParamResolverOrSimilarType) -> Dict[str, Any]:
+    """Converts a `cirq.ParamResolverOrSimilarType` to a dictionary.
+
+    Args:
+        resolver: A parameters resolver that provides values for parameter
+            references in the original `cirq.Circuit`.
+    Returns:
+        A dictionary representation of the `resolver`.
+    """
+    param_dict: Dict[str, Any] = {}
+    if isinstance(resolver, cirq.ParamResolver):
+        param_dict = resolver.param_dict
+    elif isinstance(resolver, dict):
+        param_dict = resolver
+    return param_dict
```

Comment:
This could be simplified by returning directly:

```
    if isinstance(resolver, cirq.ParamResolver):
        return resolver.param_dict
    elif isinstance(resolver, dict):
        return resolver
    return {}
```